compiler/src/main/kotlin/interpreter/interpreter.kt

```diff
@@ -0,0 +1,690 @@
+package dk.aau.cs.d409f19.cellumata.interpreter
+
+import dk.aau.cs.d409f19.cellumata.ast.*
+import dk.aau.cs.d409f19.cellumata.visitors.ASTVisitor
+import java.awt.Color
+import java.awt.Dimension
+import java.awt.Graphics2D
+import java.lang.RuntimeException
+import java.util.*
+import javax.swing.JFrame
+import javax.swing.JPanel
+
+open class InterpretRuntimeException(msg: String) : RuntimeException(msg)
+class InterpreterVisitException(node: Any) : InterpretRuntimeException("${node.javaClass} should not be visited.")
+
+/**
+ * Custom values used by the interpreter. StateValue is a state and a index (used for multi-states).
+ * When a state is mentioned in the source code, e.g. Alive, a StateValue equal to (Alive, 0) is created.
+ * The LookupExpr affects the index of the state value, e.g. Alive[2] = (Alive, 2)
+ */
+data class StateValue(val decl: StateDecl, val index: Int) {
+    override fun toString(): String {
+        return "${decl.ident}[$index]"
+    }
+}
+
+/**
+ * Special value returned by break-statements
+ */
+object BreakValue
+
+/**
+ * Special value returned by continue-statements
+ */
+object ContinueValue
+
+class Interpreter(val rootNode: RootNode) : ASTVisitor<Any> {
+
+    private val stack = MemoryStack()
+    private var defaultStateValue: StateValue? = null
+
+    fun start() {
+        visit(rootNode)
+    }
+
+    /**
+     * Fill MemoryStack with built-in functions, custom functions, neighbourhood-, and state-declarations.
+     * Constants are added last as they can depend on other declarations.
+     */
+    private fun prefillMemory(root: RootNode) {
+        stack.pushStack()
+        // Add built-in functions
+        stack.declareGlobal(BuiltinFuncCount.ident, BuiltinFuncCount)
+        stack.declareGlobal(BuiltinFuncRandi.ident, BuiltinFuncRandi)
+        stack.declareGlobal(BuiltinFuncRandf.ident, BuiltinFuncRandf)
+        stack.declareGlobal(BuiltinFuncAbsi.ident, BuiltinFuncAbsi)
+        stack.declareGlobal(BuiltinFuncAbsf.ident, BuiltinFuncAbsf)
+        stack.declareGlobal(BuiltinFuncFloor.ident, BuiltinFuncFloor)
+        stack.declareGlobal(BuiltinFuncCeil.ident, BuiltinFuncCeil)
+        stack.declareGlobal(BuiltinFuncRoot.ident, BuiltinFuncRoot)
+        stack.declareGlobal(BuiltinFuncPow.ident, BuiltinFuncPow)
+
+        // Add components except neighbourhoods. Constants are added last as they need computation
+        root.body
+            .filter { it !is ConstDecl && it !is NeighbourhoodDecl}
+            .forEach {
+                when (it) {
+                    is StateDecl -> stack.declareGlobal(it.ident, StateValue(it, 0))
+                    is FuncDecl -> stack.declareGlobal(it.ident, it)
+                    else -> AssertionError()
```

Comment:
```suggestion
                    else -> AssertionError("Received unknown root component: ${it.javaClass}")
```

Minor, but should have a proper error-message.

Reply:
Added in ca0e7fc